python/lib/modeldata/dmod/test/test_subset_service_impl.py

```diff
@@ -0,0 +1,155 @@
+import git
+import unittest
+from hypy import Catchment
+from geopandas import GeoDataFrame
+from pandas import DataFrame
+from pathlib import Path
+from typing import Dict, Optional, Union
+from ..modeldata.subset import SubsetHandlerImpl
+
+
+class TestSubsetServiceImpl(unittest.TestCase):
+
+    CAT_KEY = 'catchment_geojson_file'
+    FLOW_KEY = 'flowpath_geojson_file'
+    NEX_KEY = 'nexus_geojson_file'
+    CROSS_KEY = 'crosswalk_json_file'
+
+    @classmethod
+    def find_project_root(cls, path: Optional[Path] = None, as_path_obj: bool = False) -> Union[Path, str, None]:
```

Comment:
We really gotta figure out a way to share this function without copying it to every test module 😄

Reply:
Indeed, though FWIW I actually implemented it properly this time.  This version uses Git to determine the root, rather than guessing based on (currently valid, but not set in stone) path names and directory structure.

Reply:
Will open issue to work on formalizing this across the project